### ISSUE TITLE
Not working with Lottie-Web v5.1.13

### ISSUE THREAD

**Issue by taylorjames** [bodymovin/lottie-api#1]
I am trying to use this library to modify some lottie animations but have run into some problems where the api cannot find the data on the layers.

When I call `animationAPI.getLayersByName('text-layer')` I get `Cannot read property 'nm' of undefined`. It seems as if maybe Lottie-web changed its animation model or maybe Im not loading things properly?

Upon further inspection of LayerList.js:18, none of the elements in the loop have a `data` property which is causing it to throw an error.

Here is my code

```// SVGEngine.js
import SVG from 'svg.js';
import Lottie from 'lottie-web';
import LottieAPI from 'lottie-api';

export default class SVGEngine {
  constructor() {
    // this.height = height;
    // this.width = width;
  }

  start(id) {
    if (!SVG.supported) {
      // SVG NOT SUPPORTED
      console.log('svg not supported');
      return;
    }


    const element = document.getElementById(id);
    var animationAPI;
    var animData = {
      container: element, // the dom element that will contain the animation
      renderer: 'svg',
      loop: false,
      autoplay: true,
      path: '../static/assets/test/animation-test-01.json' // the path to the animation json
    };
    const anim = Lottie.loadAnimation(animData);

    anim.addEventListener("DOMLoaded", function(){
      animationAPI = LottieAPI.createAnimationApi(anim);

      console.log('anim api', animationAPI.getLayersByName('text-layer'));
    });



    console.log('anim',anim);
    
  }
}```

**Comment by bodymovin** [comment]
Hi, can you share the json file?

**Comment by taylorjames** [comment]
Here is a link to the .json file
https://github.com/taylorjames/lottie-test/blob/master/animation-test-01.json

I am able to play the animation fine with lottie-web

**Comment by taylorjames** [comment]
Thanks for your reply!

**Comment by taylorjames** [comment]
Thanks, I am building a web video editor in which you can add animated text layers to a video. Is there a way to easily scale/transform a layer or group of layers from the api?

**Comment by bodymovin** [comment]
Not really, the main difficulty is to animate them without javascript.
Regarding the scale layer, you can traverse the property model to the scale and pass a value to it.
I can give you an example later if you need.

**Comment by taylorjames** [comment]
As I am getting deeper into this project I have more questions. Would you be kind enough to jump on a google hangout sometime?

**Comment by taylorjames** [comment]
I found that the solution to a lot of my questions is to create expressions in AE that are then exported with your plugin. I have a few questions about the fonts. I do not want to export the characters to svg because the text in the animation is just a placeholder for user specified text.  My issue is that when I do not convert the characters, my expression that gets the size of the text does not work.

Basically, I am trying to use an expression to make a background rect the same size as the text rect and add a little padding. 

Here you can see my ae file and the exported json. 
[https://github.com/taylorjames/lottie-test](url)

Can you tell me why the background box does not resize when played with the lottie web svg player?

**Comment by taylorjames** [comment]
@bodymovin can you please advise? I can pay you for your time

**Comment by bodymovin** [comment]
Hi, sorry I've been busy and I missed this email.
Do you have a link with the implementation?
I've just tried it and it seems to work fine

**Comment by taylorjames** [comment]
weird, ok let me see if I can make something public for you to try

**Comment by taylorjames** [comment]
In after effects the yellow box behind the text is as wide as the text plus a little padding. When lottie plays it, it looks as though it is only the padding dimensions. The animation works however the yellow box is not sized correctly according to the text. Are you seeing the same result as in after effects?

Thanks bodymovin!

**Comment by bodymovin** [comment]
Yes, I'm seeing the yellow box resize together with the text dimensions.

**Comment by taylorjames** [comment]
Interesting, what implementation are you using to play the animation?

**Comment by bodymovin** [comment]
nothing out of the ordinary

**Comment by taylorjames** [comment]
I found the issue.

This works.
```
var animData = {
          container: state.element,
          renderer: 'svg',
          loop: false,
          autoplay: false,
          path: '/static/assets/test/animation-test-01.json',
          rendererSettings: {
            className:'svg-layer-'+idx
          }
        };
```

this doesnt work
```
var animData = {
          container: state.element,
          renderer: 'svg',
          loop: false,
          autoplay: false,
          animationData: layer.svg.json,
          rendererSettings: {
            className:'svg-layer-'+idx
          }
        };
```

The json in the second one is imported via webpack and passed as a key/value to the layer

**Comment by bodymovin** [comment]
I just tried it with animationData too, and it works on my side too.
Is it possible that the element is not added to the DOM when started?
getSourceRectAtTime relies on getBBox and it probably doesn't measure correctly the element if it's not yet drawn.

**Comment by taylorjames** [comment]
Maybe It could have something to do with how webpack is loading the json?

**Comment by taylorjames** [comment]
I am using react server side so that could be

**Comment by bodymovin** [comment]
I need to work on removing that dependency to measure text layers, but unfortunately it's not ready yet.

**Comment by taylorjames** [comment]
Ok, I got it working. 

The issue was the I was hiding all the layers with `display: "none"` soon as they were created.

**Comment by taylorjames** [comment]
Thanks for your help @bodymovin

**Comment by taylorjames** [comment]
@bodymovin would it be possible to create an after effects file from the exported json file?

**Comment by bodymovin** [comment]
It should be possible to a certain extent, but I haven't worked on it yet.
